src/design-system/components/checkbox/CheckBox.tsx

```diff
@@ -24,10 +31,18 @@ const CheckBox: FC<CheckBoxProps> = ({
   size = 'md',
   ...props
 }) => {
-  const [checked, setChecked] = useState<boolean>(false)
+  const [internalChecked, setInternalChecked] = useState<boolean>(false)
+
+  const isCheckboxControllable = checked !== undefined
 
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setChecked(event.target.checked)
+    const newChecked = event.target.checked
```

Comment:
We don't need event in `onChange` callback, so you can use destructurization here or even in function declaration, like `{ target: { checked } } : React.ChangeEvent<HTMLInputElement>)`